### PR TITLE
feat: transaction errors UI [NMA-1277]

### PR DESCRIPTION
**Pull request by Syn-McJ**
We want to change the UI for transactions that have errors. Both in history and transaction details.

## Issue being fixed or feature implemented
- UI is changed.
- Added "rescan blockchain" option to the TX details.
- Fixed a UI bug that prevents the "syncing" label to show up after a rescan.

## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/src/main/java/org/dash/wallet/common/util/TextViewExt.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import android.text.*
+import android.text.method.LinkMovementMethod
+import android.text.style.ClickableSpan
+import android.view.View
+import android.widget.TextView
+import androidx.annotation.ColorRes
+import androidx.core.content.ContextCompat
+import java.util.*
+
+fun TextView.makeLinks(vararg links: Pair<String, View.OnClickListener>, @ColorRes linkColor: Int = -1, isUnderlineText: Boolean = true) {
```
Comment (Syn-McJ) [inline common/src/main/java/org/dash/wallet/common/util/TextViewExt.kt:28]: This file is lifted from `dashdirect` branch.

File: wallet/src/de/schildbach/wallet/ui/SettingsActivity.kt [hunk 2]
```diff
@@ -88,17 +87,14 @@ class SettingsActivity : BaseMenuActivity() {
...
             if (it == true) {
-                isFinished = true
                 log.info("manually initiated blockchain reset")
                 analytics.logEvent(AnalyticsConstants.Settings.RESCAN_BLOCKCHAIN_RESET, bundleOf())
 
                 WalletApplication.getInstance().resetBlockchain()
                 WalletApplication.getInstance().configuration.updateLastBlockchainResetTime()
                 startActivity(WalletActivity.createIntent(this@SettingsActivity))
             } else {
-                if (!isFinished) {
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/SettingsActivity.kt:99]: Doesn't looks like `isFinished` is needed. Might be a leftover from previous refactoring.

File: wallet/src/de/schildbach/wallet/ui/main/MainViewModel.kt [hunk 3]
```diff
@@ -263,7 +263,7 @@ class MainViewModel @Inject constructor(
     }
 
     private fun updateSyncStatus(state: BlockchainState) {
-        if (_isBlockchainSyncFailed.value != state.isSynced()) {
+        if (_isBlockchainSynced.value != state.isSynced()) {
```
Comment (Syn-McJ) [inline wallet/src/de/schildbach/wallet/ui/main/MainViewModel.kt:266]: This is a bug that was preventing the "syncing" label from showing up unless the activity is relaunched.

**Review by HashEngineering — APPROVED** [review]
Great work, LGTM.

**Pull request merged**